echo-telemetry/src/main/java/com/netflix/spinnaker/echo/telemetry/TelemetryEventListener.java

```diff
@@ -160,9 +155,22 @@ public void processEvent(Event event) {
 
     Application application = Application.newBuilder().setId(hashedApplicationId).build();
 
+    DeploymentMethod.Type deployType =
+        DeploymentMethod.Type.valueOf(
+            parseEnum(
+                DeploymentMethod.Type.getDescriptor(),
+                telemetryConfigProps.getDeploymentMethod().getType().orElse("none").toUpperCase()));
+
+    DeploymentMethod deploymentMethod =
+        DeploymentMethod.newBuilder()
+            .setType(deployType)
+            .setVersion(telemetryConfigProps.getDeploymentMethod().getVersion().orElse("none"))
+            .build();
+
     SpinnakerInstance spinnakerInstance =
         SpinnakerInstance.newBuilder()
             .setId(hashedInstanceId)
+            .setDeploymentMethod(deploymentMethod)
```

Comment:
I would probably say only add a deployment method iff the Optionals from getType and getVersion are not null. Otherwise, there's no added information.